index.js

```diff
@@ -606,7 +606,14 @@ Browserify.prototype._recorder = function (opts) {
 Browserify.prototype._json = function () {
     return through.obj(function (row, enc, next) {
         if (/\.json$/.test(row.file)) {
-            row.source = 'module.exports=' + sanitize(row.source);
+            var sanitizedString = sanitize(row.source);
+            try {
+                // check json validity
+                JSON.parse(sanitizedString);
+                row.source = 'module.exports=' + sanitizedString;
+            } catch (e) {
+                this.emit('error', syntaxError(e.message, row.file || row.id));
```

Comment:
`syntaxError` here refers to the [syntax-error](https://www.npmjs.com/package/syntax-error) library, which checks if its first argument contains correct javascript syntax. So doing `syntaxError(e.message)` would check the error message for syntax errors. I think instead you can just emit the `e` value